http-server-netty/src/test/groovy/io/micronaut/http/server/netty/binding/HttpResponseSpec.groovy

```diff
@@ -200,13 +207,102 @@ class HttpResponseSpec extends AbstractMicronautSpec {
         server.close()
     }
 
+    void "test keep alive connection header is not set by default for > 499 response"() {
+        when:
+        DefaultHttpClientConfiguration config = new DefaultHttpClientConfiguration()
+        // The client will explicitly request "Connection: close" unless using a connection pool, so set it up
+        config.connectionPoolConfiguration.enabled = true
+        config.connectionPoolConfiguration.maxConnections = 2;
+        config.connectionPoolConfiguration.acquireTimeout = Duration.of(3, ChronoUnit.SECONDS);
+
+        ApplicationContext applicationContext = Micronaut.run()
+        EmbeddedServer embeddedServer = applicationContext.run(EmbeddedServer, [(SPEC_NAME_PROPERTY):getClass().simpleName])
+        RxHttpClient client = applicationContext.createBean(RxHttpClient, embeddedServer.getURL(), config)
+
+        client.exchange(
+          HttpRequest.GET('/test-header/fail')
+        ).blockingFirst()
+
+        then:
+        def e = thrown(HttpClientResponseException)
+        e.response.status == HttpStatus.INTERNAL_SERVER_ERROR
+        e.response.header(HttpHeaders.CONNECTION) == 'close'
+
+        cleanup:
+        client.stop()
+        applicationContext.stop()
+    }
+
+    void "test connection header is defaulted to keep-alive when configured to true for > 499 response"() {
+        when:
+        DefaultHttpClientConfiguration config = new DefaultHttpClientConfiguration()
+        // The client will explicitly request "Connection: close" unless using a connection pool, so set it up
+        config.connectionPoolConfiguration.enabled = true
+        config.connectionPoolConfiguration.maxConnections = 2;
+        config.connectionPoolConfiguration.acquireTimeout = Duration.of(3, ChronoUnit.SECONDS);
+
+        ApplicationContext applicationContext = Micronaut.run()
+        EmbeddedServer embeddedServer = applicationContext.run(EmbeddedServer, [
+          (SPEC_NAME_PROPERTY):getClass().simpleName,
+          'micronaut.server.netty.keepAliveOnServerError':true
+        ])
+        RxHttpClient client = applicationContext.createBean(RxHttpClient, embeddedServer.getURL(), config)
+
+        client.exchange(
+          HttpRequest.GET('/test-header/fail')
+        ).blockingFirst()
+
+        then:
+        def e = thrown(HttpClientResponseException)
+        e.response.status == HttpStatus.INTERNAL_SERVER_ERROR
+        e.response.header(HttpHeaders.CONNECTION) == 'keep-alive'
+
+        cleanup:
+        client.stop()
+        applicationContext.stop()
+    }
+
+    void "test connection header is not defaulted to keep-alive  when configured to false for > 499 response"() {
```

Comment:
I don't think this test adds value. Specifying the default and not specifying is the same thing

Reply:
removed this one.